File: 백준/Silver/2606. 바이러스/바이러스.py
n=int(input()) # 컴퓨터 개수
v=int(input()) # 연결선 개수

graph = [[] for _ in range(n + 1)] # 그래프 초기화
visited=[False] * (n + 1) # 방문한 컴퓨터인지 표시

for i in range(v): # 그래프 생성
    a,b=map(int,input().split())
    graph[a].append(b) 
    graph[b].append(a) 

def dfs(start):
    visited[start]=True
    for next in graph[start]:
        if not visited[next]:
            dfs(next)
dfs(1)
print(sum(visited)-1)

File: 백준/Silver/2606. 바이러스/바이러스.java
import java.io.*;
import java.util.*;
public class Main {
    static int map[][];
    static boolean visited[];
    static int N;
    static int M;
    static int result = 0;


    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        N = Integer.parseInt(br.readLine());
        M = Integer.parseInt(br.readLine());

        visited = new boolean[N + 1];
        map = new int[N + 1][N + 1];

        for (int i = 0; i < M; i++) {
            StringTokenizer st = new StringTokenizer(br.readLine());

            int a = Integer.parseInt(st.nextToken());
            int b = Integer.parseInt(st.nextToken());

            map[a][b] = map[b][a] = 1;
        }

        dfs(1);
        System.out.println(result - 1);
    }

    public static void dfs(int start) {
        if (visited[start]) {
            return;
        }
        visited[start] = true;
        result++;
        
        for (int i = 1; i < N + 1; i++) {
            if (map[start][i] == 1 && !visited[i]) {
                dfs(i);
            }
        }
    }
}

